src/averageAges.js

```diff
@@ -14,12 +14,21 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateMenAverageAge(people, century = 0) {
+  if (century === 0) {
+    const menArray = people.filter(men => men.sex === 'm');
+    const yearsSum = menArray.reduce((years, men) =>
+      years + (men.died - men.born), 0);
+
+    return yearsSum / menArray.length;
+  } else {
+    const menFromCentury = people.filter(men =>
+      men.sex === 'm' && Math.ceil(men.died / 100) === century);
+    const yearsSum = menFromCentury.reduce((years, men) =>
+      years + (men.died - men.born), 0);
+
+    return yearsSum / menFromCentury.length;
```

Comment:
This logic is repeated in every function. Let's put this in a separate function `calculateAverageAge` that will take an array of numbers and a total number